scrapper.py

```diff
@@ -23,75 +36,310 @@ class IncorrectNumberOfArticlesError(Exception):
+        os.makedirs(constants.PAGES_ROOT)
+
+    if not os.path.exists(constants.ASSETS_PATH):
+        os.makedirs(constants.ASSETS_PATH)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        config = json.load(file)
+
+    is_config_a_dict = isinstance(config, dict)
+
+    has_config_attributes = (
+        'base_urls' in config and
+        'total_articles_to_find_and_parse' in config
+    )
+
+    if not (is_config_a_dict and has_config_attributes):
+        raise UnknownConfigError
+
+    if 'max_number_articles_to_get_from_one_seed' not in config:
+        config['max_number_articles_to_get_from_one_seed'] = \
+            config['total_articles_to_find_and_parse']
+
+    is_base_urls_correct = (
+            isinstance(config['base_urls'], list) and
+            all([isinstance(seed_url, str) for seed_url in config['base_urls']])
+    )
+
+    if isinstance(config['total_articles_to_find_and_parse'], int):
+        is_total_number_of_articles_correct = config['total_articles_to_find_and_parse'] > 0
+    else:
+        raise IncorrectNumberOfArticlesError
+
+    is_max_number_of_articles_int = (
+        isinstance(config['max_number_articles_to_get_from_one_seed'], int) and
+        config['max_number_articles_to_get_from_one_seed'] > 0
+    )
+
+    is_max_number_of_articles_correct = (
+            config['max_number_articles_to_get_from_one_seed'] <= 10000
+    )
+
+    if not is_base_urls_correct:
+        raise IncorrectURLError
+
+    if not is_total_number_of_articles_correct:
+        raise IncorrectNumberOfArticlesError
+
+    if not is_max_number_of_articles_correct:
+        raise NumberOfArticlesOutOfRangeError
+
+    if all((
+            is_config_a_dict,
+            has_config_attributes,
+            is_base_urls_correct,
+            is_total_number_of_articles_correct,
+            is_max_number_of_articles_int,
+            is_max_number_of_articles_correct
+    )):
+        return config.values()
+
+    raise UnknownConfigError
+
+
+def load_previous_state(crawler_obj):
+    if os.path.exists(constants.TO_PARSE_URLS):
+        crawler_obj.seed_urls = open(constants.TO_PARSE_URLS).read().split('\n')
+
+    if os.path.exists(constants.SEEN_URLS):
+        crawler_obj.seen_urls = set(open(constants.SEEN_URLS).read().split('\n'))
+
+    if os.path.exists(constants.ARTICLE_URLS):
+        crawler_obj.urls = open(constants.ARTICLE_URLS).read().split('\n')
+    return crawler_obj
 
 
 if __name__ == '__main__':
-    # YOUR CODE HERE
-    pass
+    prepare_environment()
+
+    try:
+        urls, articles_max, articles_per_seed = validate_config(constants.CRAWLER_CONFIG_PATH)
+    except (
+            IncorrectURLError,
+            IncorrectNumberOfArticlesError,
+            NumberOfArticlesOutOfRangeError,
+            UnknownConfigError
+    ) as error:
+        print(f'{error} occurred')
```

Comment:
the program should finish with a non-zero code

Reply:
fixed